javascript/javascript1/week3/cactusIoInteractive.js

```diff
@@ -0,0 +1,61 @@
+/*
+ * Javascript file for Hack Your Future, Javascript1, homework Week3
+ * program will help users manage their smart phone usage.
+*/
+
+// CactusIO-interactive (Smart phone usage app)
+
+// Adding an activity
+
+const activities = [];
+var today = new Date().toLocaleDateString("en-UE");
+function addActivity(date, activity, duration) {
+    activities.push({ date, activity, duration });
+};
+
+addActivity(today, "Youtube", 30);
+addActivity("29/1-21", "Instagram", 60);
+addActivity(today, "Playing games", 50);
+//addActivity(today, "Facebook", 40); // I added this line to check the time limit reach for today
+
+console.log(activities);
+
+// Show my status
+
+function showStatus(activityArray, limitTime = 100) {
+    let totalDuration = 0;
+    let todayDuration = 0;
+    let todayActivityNumbers = 0;
+    for (let i = 0; i < activityArray.length; i++) {
+        totalDuration += activityArray[i].duration;
+        if (activityArray[i].date === today) {
+            todayDuration += activityArray[i].duration;
+            todayActivityNumbers++
+        }
+    }
+
+    if (!activityArray || !activityArray.length) {
```

Comment:
I think the `!activityArray` check is too late. If `activityArray = null` then your code would have already failed at line 29. You could remove that check or decide if you should check at the top of the `showStatus` function.

Reply:
it is done. thanks